internal/common/common.go

```diff
@@ -33,17 +49,44 @@ func CheckForVersionUpdate() {
 }
 
 // versionCheck compares local to remote version
-func versionCheck() (res *latest.CheckResponse, skip bool) {
-	githubTag := &latest.GithubTag{
-		Owner:             "kubefirst",
-		Repository:        "kubefirst",
-		FixVersionStrFunc: latest.DeleteFrontV(),
-	}
-	res, err := latest.Check(githubTag, strings.Replace(configs.K1Version, "v", "", 1))
+func versionCheck() (res *CheckResponse, skip bool) {
+	var latestVersion string
+
+	resp, err := http.Get("https://raw.githubusercontent.com/Homebrew/homebrew-core/master/Formula/k/kubefirst.rb")
+
 	if err != nil {
-		fmt.Printf("checking for a newer version failed with: %s", err)
+		fmt.Printf("checking for a newer version failed (cannot get Homebrew formula) with: %s", err)
+		return nil, true
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		bodyBytes, err := io.ReadAll(resp.Body)
+
+		if err != nil {
+			fmt.Printf("checking for a newer version failed (cannot read the file) with: %s", err)
+			return nil, true
+		}
+
+		bodyString := string(bodyBytes)
+		if strings.Contains(bodyString, "url \"https://github.com/kubefirst/kubefirst/archive/refs/tags/") {
+			re := regexp.MustCompile(`.*/v(.*).tar.gz"`)
+			matches := re.FindStringSubmatch(bodyString)
+			latestVersion = matches[1]
+			fmt.Printf(latestVersion)
+		} else {
+			fmt.Printf("checking for a newer version failed (no reference to kubefirst release) with: %s", err)
+			return nil, true
+		}
+	} else {
+		fmt.Printf("checking for a newer version failed (HTTP error) with: %s", err)
 		return nil, true
 	}
```

Comment:
What you have is plenty good. Here is another way to write it such that you handle the errors and then proceed executing code without having to wrap it in and if condition. Faster in some cases but also more readable and flat. 
```suggestion
if resp.StatusCode != http.StatusOK {
		fmt.Printf("checking for a newer version failed (HTTP error) with: %s", err)
		return nil, true
	}

	bodyBytes, err := io.ReadAll(resp.Body)
	if err != nil {
		fmt.Printf("checking for a newer version failed (cannot read the file) with: %s", err)
		return nil, true
	}

	bodyString := string(bodyBytes)

	if !strings.Contains(bodyString, "url \"https://github.com/kubefirst/kubefirst/archive/refs/tags/") {
		fmt.Printf("checking for a newer version failed (no reference to kubefirst release) with: %s", err)
		return nil, true
	}
	
	re := regexp.MustCompile(`.*/v(.*).tar.gz"`)
	matches := re.FindStringSubmatch(bodyString)
	latestVersion = matches[1]
	fmt.Printf(latestVersion)
```

Reply:
Is it a best practice in Go? I'm like a junior Go developer, it's fun 😄

In other languages, I wouldn't do that many return, nor would isolate the error handling.